lib/common.ts

```diff
@@ -0,0 +1,48 @@
+import { BalenaSDK } from 'balena-sdk';
+import { ExpectedError } from './errors';
+
+// TODO fix this any
+export function getReleaseTagKeysAndValues(options: any) {
+	const releaseTags = options['release-tag'] ?? [];
+	const releaseTagKeys = releaseTags.filter(
+		(_v: any, i: number) => i % 2 === 0,
+	);
+	const releaseTagValues = releaseTags.filter(
+		(_v: any, i: number) => i % 2 === 1,
+	);
+
+	releaseTagKeys.forEach((key: string) => {
+		if (key === '') {
+			throw new ExpectedError(`Error: --release-tag keys cannot be empty`);
+		}
+		if (/\s/.test(key)) {
+			throw new ExpectedError(
+				`Error: --release-tag keys cannot contain whitespaces`,
+			);
+		}
+	});
+	if (releaseTagKeys.length !== releaseTagValues.length) {
+		releaseTagValues.push('');
+	}
+	return { releaseTagKeys, releaseTagValues };
+}
+
+export async function applyReleaseTagKeysAndValues(
+	sdk: BalenaSDK,
+	releaseId: number,
+	releaseTagKeys: string[],
+	releaseTagValues: string[],
+) {
+	if (releaseTagKeys.length === 0) {
+		return;
+	}
+	// Above we have checked that releaseTagKeys and releaseTagValues are of the same size
```

Comment:
This comment could be moved to a function documentation block (`/** ... */`), to specify a contract that the caller must abide to when calling this function (pompous way of saying 😁  that it is up to the caller to ensure that `releaseTagKeys` and `releaseTagValues` arguments are of the same length).